params/hooks_libevm.go

```diff
@@ -115,7 +131,7 @@ func (r RulesExtra) precompileOverrideBuiltin(addr common.Address) (libevm.Preco
 		return nil, false
 	}
 
-	return makePrecompile(precompile), true
```

Comment:
Removing this forces every line of the map-literals to use it. What is the advantage of it not just being done once, here?

Reply:
It looks like it's because of the mix of `vm.PrecompiledContract` (the one we're registering) and `contract.StatefulPrecompiledContract` (everything else) types. It's messy but I'm not sure how we can get around it without a major refactor elsewhere.